pkg/controller/backup_controller_test.go

```diff
@@ -422,3 +426,150 @@ func TestProcessBackup(t *testing.T) {
+			backup:           arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew),
+			defaultLocations: defaultLocationAWS,
+			expectedSuccess:  true,
+		},
+		{
+			name:            "no existing location name and no default location name given",
+			backup:          arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew),
+			expectedSuccess: true,
+		},
+		{
+			name:             "multiple location names for a provider, default location name for another provider",
+			backup:           arktest.NewTestBackup().WithName("backup1").WithPhase(v1.BackupPhaseNew).WithVolumeSnapshotLocations(dupLocationNames),
+			locations:        arktest.NewTestVolumeSnapshotLocation().WithName(dupLocationNames[0]).WithProviderConfig(dupLocationList),
+			defaultLocations: defaultLocationGCP,
+			expectedSuccess:  true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				client          = fake.NewSimpleClientset()
+				backupper       = &fakeBackupper{}
+				sharedInformers = informers.NewSharedInformerFactory(client, 0)
+				logger          = logging.DefaultLogger(logrus.DebugLevel)
+				pluginManager   = &pluginmocks.Manager{}
+				backupStore     = &persistencemocks.BackupStore{}
+			)
+			defer backupper.AssertExpectations(t)
+			defer pluginManager.AssertExpectations(t)
+			defer backupStore.AssertExpectations(t)
+
+			c := NewBackupController(
+				sharedInformers.Ark().V1().Backups(),
+				client.ArkV1(),
+				backupper,
+				false,
+				logger,
+				logrus.InfoLevel,
+				func(logrus.FieldLogger) plugin.Manager { return pluginManager },
+				NewBackupTracker(),
+				sharedInformers.Ark().V1().BackupStorageLocations(),
+				"default",
+				sharedInformers.Ark().V1().VolumeSnapshotLocations(),
+				test.defaultLocations,
+				metrics.NewServerMetrics(),
+			).(*backupController)
+
+			c.newBackupStore = func(*v1.BackupStorageLocation, persistence.ObjectStoreGetter, logrus.FieldLogger) (persistence.BackupStore, error) {
+				return backupStore, nil
+			}
+
+			// set up a Backup object to represent what we expect to be passed to backupper.Backup()
+			backup := test.backup.DeepCopy()
+			backup.Spec.VolumeSnapshotLocations = test.backup.Spec.VolumeSnapshotLocations
+			for _, l := range test.locations {
+				location := l.DeepCopy()
+				location.Name = l.Name
+				location.Spec.Provider = l.Spec.Provider
+				location.Spec.Config = l.Spec.Config
+
+				vsl := &v1.VolumeSnapshotLocation{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: backup.Namespace,
+						Name:      location.Name,
+					},
+					Spec: v1.VolumeSnapshotLocationSpec{
+						Provider: location.Spec.Provider,
+						Config:   location.Spec.Config,
+					},
+				}
+				require.NoError(t, sharedInformers.Ark().V1().VolumeSnapshotLocations().Informer().GetStore().Add(vsl))
+			}
+
+			errs := c.defaultAndValidateSnapshotLocations(backup, test.defaultLocations)
```

Comment:
It would be nice to also have the test validate that `test.backup.Spec.VolumeSnapshotLocations` and `test.backup.Status.ValidationErrors` contain what you expect.

Reply:
Good point, added.